### ISSUE TITLE
Best model in ModelCheckpoint

### ISSUE THREAD

**Issue by mpariente** [PyTorchLightning/pytorch-lightning#1960]
## 🚀 Feature
Add a `best_model` attribute in `ModelCheckpoint`. 

### Motivation
After training, it would be nice to easily have the path to the checkpoint with the best val loss. 

Or is there an argument in the trainer to resume best state after training and I missed it? (totally possible)

**Comment by mpariente** [comment]
BTW, this is my solution but I don't think this should be necessary 
```python
best_k = checkpoint.best_k_models
best_path = [b for b, v in best_k.items() if v == torch.min(best_k.values())][0]
```

**Comment by rohitgr7** [comment]
@mpariente Better condition will be:
```python
best_k = checkpoint.best_k_models
best_path = [b for b, v in best_k.items() if v == checkpoint.best)][0]
```
`checkpoint.mode`can be `max` though.

**Comment by mpariente** [comment]
True, thanks, but this still doesn't change the fact that a `best_model_path` attribute should be integrated in the checkpoint IMO

**Comment by HansBambel** [comment]
The `save_top_k` parameter saves the last k best models. When put to -1 it saves all models. The latest master version also has a `save_last` parameter that also keeps the latest epoch of the model.
E.g:
```
checkpoint_callback = ModelCheckpoint(
        filepath=os.getcwd()+"/"+"<your-folder>"+"/{epoch}-{val_loss:.6f}",
        save_top_k=1,
        verbose=False,
        monitor='val_loss',
        mode='min',
        prefix=net.__class__.__name__+"_"
    )
```
This results in a file called: `UNet_epoch=5-val_loss=0.581735`

**Comment by mpariente** [comment]
Thanks for the example, I also use it like that. 
Still, asking the user to reverse the dict or look for the best loss in the filenames is not super user-friendly. It asks few (not many) additional lines of code which are not necessary, don't you think?

**Comment by williamFalcon** [comment]
this is already in master

**Comment by mpariente** [comment]
Oh ok, sorry for bothering then, I should have checked master 
Thanks for the pointer William

**Comment by williamFalcon** [comment]
it's a great idea :) just someone beat you to it haha
